src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,876 @@
+
+@doc raw"""
+Tucker{N, R, D, 𝔽} <: AbstractManifold{𝔽}
+
+The manifold of $N_1 \times \dots \times N_D$ real-valued or complex-valued tensors of
+fixed multilinear rank $(R_1, \dots, R_D)$ . If $R_1 = \dots = R_D = 1$, this is the
+manifold of rank-1 tensors.
+
+# Representation in HOSVD format
+
+Any tensor $\mathcal{A}$ on the Tucker manifold can be represented in HOSVD
+[^DeLathauwer2000] form
+```math
+\mathcal{A} = (U_1,\dots,U_D) \cdot \mathcal{C}
+```
+where $\mathcal C \in \mathbb{F}^{R_1 \times \dots \times R_D}$ and, for $d=1,\dots,D$,
+the matrix $U_d \in \mathbb{F}^{N_d \times R_d}$ contains the singular vectors of the
+$d$th unfolding of $\mathcal{A}$
+
+# Tangent space
+
+The tangent space to the Tucker manifold at
+$\mathcal{A} = (U_1,\dots,U_D) \cdot \mathcal{C}$ is [^Koch2010]
+```math
+T_{\mathcal{A}} \mathcal{M} =
+\bigl\{
+(U_1,\dots,U_D) \cdot \dot{\mathcal{C}}
++ \sum_{d=1}^D \bigl(
+    (U_1, \dots, U_{d-1}, \dot{U}_d, U_{d+1}, \dots, U_D)
+    \cdot \mathcal{C}
+\bigr)
+\bigr\}
+```
+where $\dot{\mathcal{C}}$ is arbitrary, $\dot{U}_d^{\mathrm{H}}$ is the Hermitian adjoint of
+$U_d$, and $\dot{U}_d^{\mathrm{H}} U_d = 0$ for all $d$.
+
+# Constructor
+    Tucker(n⃗::NTuple{D, Int}, r⃗::NTuple{D, Int}[, field = ℝ])
```

Comment:
It would be great if all input variables could shortly be explained in the text after this line.